verilog/analysis/checkers/struct_union_name_style_rule.cc

```diff
@@ -79,15 +82,65 @@ void StructUnionNameStyleRule::HandleSymbol(const verible::Symbol& symbol,
       // Neither a struct nor union definition
       return;
     }
-    const auto* identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
+    const auto *identifier_leaf = GetIdentifierFromTypeDeclaration(symbol);
     const auto name = ABSL_DIE_IF_NULL(identifier_leaf)->get().text();
-    if (!verible::IsLowerSnakeCaseWithDigits(name) ||
+    // get rid of the exceptions specified by user
+    std::string name_filtered;
+    if (!exceptions_.empty()) {
+      name_filtered.reserve(name.length());
+      const auto &underscore_separated_parts = absl::StrSplit(name, '_');
+      if (name[0] != '_') {
+        for (const auto &ns : underscore_separated_parts) {
+          if (std::find(exceptions_.begin(), exceptions_.end(), ns) ==
```

Comment:
Note: Finding things in a set is easier: you can do `exceptions.find(...) != exceptions.end()` - which is a fast `O(log N)` tree-lookup (std::set) or `O(1)` hash-lookup (std::unordered_set).

Using `std::find` essentially forces it iterate through the whole set, making it an `O(n)` operation; this is only needed for containers that don't support direct look-ups.